controller/linodemachine_controller.go

```diff
@@ -117,17 +137,151 @@ func (r *LinodeMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 			LinodeMachine: linodeMachine,
 		},
 	)
+	if err != nil {
+		log.Info("Failed to create machine scope", "error", err.Error())
+
+		return ctrl.Result{}, fmt.Errorf("failed to create machine scope: %w", err)
+	}
 
-	return r.reconcile(ctx, machineScope, clusterScope)
+	return r.reconcile(ctx, machineScope, clusterScope, linodeMachine, log)
 }
 
 func (r *LinodeMachineReconciler) reconcile(
 	ctx context.Context,
 	machineScope *scope.MachineScope,
 	clusterScope *scope.ClusterScope,
-) (ctrl.Result, error) {
-	// TODO
-	return ctrl.Result{}, nil
+	linodeMachine *infrav1.LinodeMachine,
+	logger logr.Logger,
+) (res ctrl.Result, err error) {
+	defer func() {
+		if patchErr := machineScope.PatchHelper.Patch(ctx, linodeMachine); patchErr != nil &&
+			client.IgnoreNotFound(patchErr) != nil {
+
+			logger.Error(patchErr, "failed to patch linodeMachine")
+
+			err = errors.Join(err, patchErr)
+		}
+	}()
+
+	// Delete
+	if !linodeMachine.ObjectMeta.DeletionTimestamp.IsZero() {
+		logger.Info("deleting machine")
+
+		if machineScope.LinodeMachine.GetInstanceID() != 0 {
+			if err = machineScope.LinodeClient.DeleteInstance(ctx, machineScope.LinodeMachine.GetInstanceID()); err != nil {
+				logger.Info("Failed to delete Linode machine instance", "error", err.Error())
+
+				// Not found is not an error
+				if apiErr, ok := err.(*linodego.Error); ok && apiErr.Code != http.StatusNotFound {
```

Comment:
I'm not sure this is the best way to validate linode api errors

Reply:
this is pretty typical for what ive seen